plugins/GenerateNativeScriptEntryPointsPlugin.js

```diff
@@ -39,26 +39,33 @@ exports.GenerateNativeScriptEntryPointsPlugin = (function () {
     }
 
     GenerateNativeScriptEntryPointsPlugin.prototype.generateEntryFile = function (compilation, entryPoint) {
-        const entryPointFileName = `${entryPoint.options.name}.js`;
-        if (entryPointFileName === SNAPSHOT_ENTRY_MODULE) {
+        const entryPointName = entryPoint.options.name;
+        let entryChunk;
+        if (entryPointName === SNAPSHOT_ENTRY_NAME) {
             // Do not require the snapshot entry dependencies as the snapshot will fail.
             return;
         }
 
         const requireDeps =
             entryPoint.chunks.map(chunk => {
                 let requireChunkFiles = "";
-                chunk.files.forEach(fileName => {
-                    if (fileName !== entryPointFileName) {
+                if (chunk.name === entryPointName) {
+                    entryChunk = chunk;
+                } else {
+                    chunk.files.forEach(fileName => {
                         requireChunkFiles += `require("./${fileName}");`;
-                    }
-                });
+                    });
+                }
 
                 return requireChunkFiles;
             }).join("\n");
 
-        const currentEntryPointContent = compilation.assets[entryPointFileName].source();
-        compilation.assets[entryPointFileName] = new RawSource(`${requireDeps}${currentEntryPointContent}`);
+        if (entryChunk) {
```

Comment:
it shouldn't be possible to get here without entryChunk, so I suggest the following:
```JavaScript
if (!entryChunk) {
    throw new Error(`Unable to find entryChunk named ${entryPointName} in the current entryPoint chunks: ${entryPoint.chunks.map(c => c.name).join(", ")}`);
}